Add docs for `@bodyRoot` vs `@body` (#6927)
microsoft/typespec: website/src/content/docs/docs/libraries/http/operations.md

}
```

### `@body` vs `@bodyRoot`

> The `@body` decorator defines the property which type will be exactly the body. Annotating a property inside a `@body` with applicable metadata (`@header`, `@path`, `@query` for request and `@header`, `@statusCode` for response) will be ignored and log a warning.

Reviewer: These are somewhat difficult to describe, let me know what you think of this wording.

> The `@bodyRoot` decorator defines the property where the implicit body resolution will be applied(By default the root of the operation).

Reviewer: ```suggestion
The `@bodyRoot` decorator similarly applies to a property or parameter.  The type of that decorated property or parameter is similarly used to define the request or response body.  If the body type is not a model, the behavior is identical to `@body`. If the body type is a model, instead of exactly defining *only* the body, the model may also contain properties annotated as applicable http metadata. Such properties will be treated as http metadata, will not be included in the request or response body, and will not result in a warning.
```

> Nesting `@body` and `@bodyRoot` while mostly pointless can happen when using templates to build operations. A warning will be emitted if nesting happens inline.

Reviewer: ```suggestion
Nesting `@body` and `@bodyRoot`, while mostly pointless, can happen when using templates to build operations. A warning will be emitted if nesting happens inline.
```

The meaning when nesting them is as follow:

- As soon as `@body` is reached the content is exactly the body which means any nested `@bodyRoot` or `@body` will be ignored.

> - When reaching a `@bodyRoot` it will keep looking for `@body` and `@bodyRoot` nested and if found the deepest one will be used as the body.

Reviewer: ```suggestion
- if `@bodyRoot` is reached before any occurrence of `@body`, it will keep looking for nested properties decorated with  `@body` or `@bodyRoot` and if found the deepest one will be used to determine the body.
```

Examples

<table>
<tr>
  <th>Code</th>
  <th>Example Payload</th>
</tr>
<tr>
  <td>

```typespec
op case1(
  @header foo: string, //
  name: string,
  age: int32,
): void;
```

  </td>
  <td>

```http
POST /
Foo: bar
{
  "name": "Rex",
  "age": 3
}
```

  </td>
```
  </td>
</tr>
<tr>
  <td>

```typespec
op case2(
  body: {
    @header foo: string;
    name: string;
    age: int32;
  },
): void;
```

  </td>
  <td>

```http
POST /
Foo: bar
{
  "body": {
    "name": "Rex",
    "age": 3
  }
}
```

  </td>
</tr>
<tr>
  <td>

```typespec
op case3(
  @body body: {
    @header foo: string; // warning: `@header` is ignored
    name: string;
    age: int32;
  },
): void;
```

  </td>
  <td>

```http
POST /
{
  "name": "Rex",
  "age": 3
}
```

  </td>
</tr>
<tr>
  <td>

```typespec
op case4(
  @bodyRoot body: {

>     @header foo: string;

Reviewer: ```suggestion
    @header foo: string; // property treated as a request header parameter

Author: ```suggestion
    // property treated as a request header parameter
    @header foo: string;
```

Author: kept it like that because otherwise in the table it looks weird when cells are too long. 
and if I had the comment above the formatter just separate all lines.

    name: string;
    age: int32;
  },
): void;
```

  </td>
  <td>

```http
POST /
Foo: bar
{
  "name": "Rex",
  "age": 3
}
```

  </td>
</tr>
<tr>
  <td>

```typespec
op case5(
  // This bodyRoot is a noop and will log a warning
  @bodyRoot body: {
    @bodyRoot reallyBody: {

>       @header foo: string;

Reviewer: ```suggestion
      @header foo: string; // property treated as a request header parameter
```

Author: same here
